src/cargo/sources/registry/index.rs

```diff
@@ -855,7 +858,7 @@ impl IndexSummary {
             }
         }
         let mut summary = Summary::new(config, pkgid, deps, &features, links)?;
-        summary.set_checksum(cksum);
+        cksum.map(|cksum| summary.set_checksum(cksum));
```

Comment:
Please use an explicit `if let Some()` rather than `map` for performing a side effect.

At least for myself, I see a `map` and I expect a code transformation to occur and can safely ignore that section of code as it shouldn't be mutating anything else but it is here.